src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,196 @@
+import { FlatList, PermissionsAndroid } from 'react-native'
+import { promisify } from 'es6-promisify'
+import contacts from 'react-native-contacts'
+import { map, memoize, orderBy, uniq } from 'lodash'
+import { isAndroid } from '../../lib/utils/platform'
+import { Section } from '../common'
+import Separator from '../common/layout/Separator'
+import InputText from '../common/form/InputText'
+import { withStyles } from '../../lib/styles'
+import normalize from '../../lib/utils/normalizeText'
+import { getDesignRelativeHeight } from '../../lib/utils/sizes'
+import userStorage from '../../lib/gundb/UserStorage'
+import FeedContactItem from './FeedContactItem'
+const Contacts = promisify(contacts.getAll)
+
+const WhoContent = ({ styles, setContact, error, text, value, next, state, showNext, setValue }) => {
+  const [contacts, setContacts] = React.useState([])
+  const [initialList, setInitalList] = React.useState(contacts)
+  const [recentFeedItems, setRecentFeedItems] = React.useState([])
+  const [recentlyUsedList, setRecentlyUsedList] = React.useState([])
+
+  const getUserFeed = async () => {
+    const userFeed = await userStorage.getFeedPage(5, true)
+    const recent = userFeed.filter(({ type }) => type === 'send' || type === 'receive')
+    setRecentFeedItems(recent)
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      }
+      Contacts().then((contacts, err) => {
+        if (err === 'denied') {
+          console.warn('permissions denied')
+        } else {
+          const sortContacts = orderBy(contacts, ['givenName'])
+          setContacts(sortContacts)
+          setInitalList(sortContacts)
+        }
+      })
+    }
+  }, [])
+
+  const filterContacts = useCallback(
+    memoize(searchQuery => {
+      const query = searchQuery.toLocaleLowerCase()
+
+      return initialList.filter(({ givenName, familyName, phoneNumbers }) => {
+        const [phoneNumber] = map(phoneNumbers, 'number')
+
+        return [givenName, familyName, phoneNumber].some(field => (field || '').toLocaleLowerCase().includes(query))
+      })
+    }),
+    [initialList]
+  )
+
+  const handleSearch = useCallback(
+    query => {
+      if (state) {
+        setValue(query)
+      }
+
+      setContacts(filterContacts(query))
+    },
+    [setContacts, setValue, filterContacts]
+  )
+
+  useEffect(() => {
+    if (contacts.length === 0) {
+      showNext(true)
+    } else {
+      showNext(false)
+    }
+  }, [contacts])
+
+  useEffect(() => {
+    findRecentlyUsed()
+  }, [contacts, recentFeedItems])
+
+  const findRecentlyUsed = () => {
```

Comment:
it it's used in the effect above only you could move this function inside its body